Microsoft.XmlSerializer.Generator 2.0.0 - Could not load System.Runtime 4.2.1.0

[EDIT] Investigation Completed.

Two additional issues have been identified from the investigation.
dotnet/runtime#597
dotnet/runtime#630

We will leave this issue open and update it as they get fixed.
[See down below](https://github.com/dotnet/corefx/issues/40730#issuecomment-558848416) for a workaround.

*********************************************************************************************************
Hi there!

I'm trying to generate an xml serialization assembly but I keep on stranding with the following:

> SGEN : warning SGEN1: Could not load file or assembly 'System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL'.

I've seen several posts pointing to edit the configuration file of Visual Studio but that was no solution in my case.

I tried both version 1.0.0 and 2.0.0, neither of them work, however version 1.0.0 doesn't generate any specific log output. Both versions end their logging with similar lines;

> Assembly '[path to project]\obj\Debug\netcoreapp2.2\library.dll' does not contain any types that can be serialized using XmlSerializer.
1>%USERPROFILE%\.nuget\packages\microsoft.xmlserializer.generator\2.0.0\build\Microsoft.XmlSerializer.Generator.targets(16,5): warning : SGEN: Fail to generate the serializer for library.dll. Please follow the instructions at https://go.microsoft.com/fwlink/?linkid=858594 and try again.

Interestingly enough, though, is that I got no issue on another device. This proofs that it is a local issue. I'd still like to be able to use the generator on my main device, what am I missing that this does not work?

I'm trying to connect to SQL Server Reporting Services to generate multiple PDF files (Invoices) without needing to display any UI - so just need the service connection to ReportExecution2005.asmx.  I have code that works fine in a Net Framework WinForms project, but cannot get it to work with Net Core 3.1 in a WPF project.

I've tried to use the method suggested at https://docs.microsoft.com/en-us/dotnet/core/additional-tools/xml-serializer-generator , but these instructions cannot be followed verbatim in a Net Core 3.1 WPF project (are there any specific instructions for such?).

Despite trying ALL of the suggested solutions I can find, I cannot get to create an xmlserializers.dll at build (see Rebuild output lower down), and at runtime obviously get the exception below:

======== in 'reference.cs ' ===========
[System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Advanced)]
System.Threading.Tasks.Task<REService.LoadReportResponse> REService.ReportExecutionServiceSoap.LoadReportAsync(REService.LoadReportRequest request)
{
    return base.Channel.LoadReportAsync(request); // THROWS EXCEPTION: System.IO.FileNotFoundException: 'Could not load file or assembly 'C:\Users\??\source\repos\????????\bin\Release\netcoreapp3.1\????????.XmlSerializers.dll'.
}


2>------ Rebuild All started: Project: ????????, Configuration: Release Any CPU ------
2>???????? -> C:\Users\??\source\repos\????????\bin\Release\netcoreapp3.1\????????.dll
2>[Microsoft (R) .NET Core Xml Serialization Generation Utility, Version 2.2.1]
2>Copyright (C) Microsoft Corporation. All rights reserved.
2>SGEN : warning SGEN1: Could not load file or assembly 'System.ServiceModel.Primitives, Version=4.7.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
2>SGEN : warning SGEN1: Cannot load a reference assembly for execution.
2>SGEN : warning SGEN1: Could not load file or assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
2>SGEN : warning SGEN1: Cannot load a reference assembly for execution.
2>SGEN : warning SGEN1: Could not load file or assembly 'Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
2>SGEN : warning SGEN1: Cannot load a reference assembly for execution.
2>A BadImageFormatException has been thrown while parsing the signature. This is likely due to lack of a generic context. Ensure genericTypeArguments and genericMethodArguments are provided and contain enough context.
2>Could not load file or assembly 'System.ServiceModel.Primitives, Version=4.7.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
2>Cannot load a reference assembly for execution.
2>C:\Users\??\.nuget\packages\microsoft.xmlserializer.generator\2.2.1\build\Microsoft.XmlSerializer.Generator.targets(37,5): warning MSB3073: The command "dotnet Microsoft.XmlSerializer.Generator "obj\Release\netcoreapp3.1\????????.dll" --force --quiet obj\Release\netcoreapp3.1\sgen.rsp" exited with code 1.
2>C:\Users\??\.nuget\packages\microsoft.xmlserializer.generator\2.2.1\build\Microsoft.XmlSerializer.Generator.targets(38,5): warning : SGEN: Failed to generate the serializer for ????????.dll. Please follow the instructions at https://go.microsoft.com/fwlink/?linkid=858594 and try again.
2>Done building project "????????.csproj".
========== Rebuild All: 2 succeeded, 0 failed, 0 skipped ==========



========= Contents of 'ConnectedService.json' (and reference classes successfully generated when adding service connection to project) ===========
{
  "ProviderId": "Microsoft.VisualStudio.ConnectedService.Wcf",
  "Version": "15.0.40203.910",
  "GettingStartedDocument": {
    "Uri": "https://go.microsoft.com/fwlink/?linkid=858517"
  },
  "ExtendedData": {
    "inputs": [
      "http://??.??.??.??/ReportServer/ReportExecution2005.asmx"
    ],
    "collectionTypes": [
      "System.Array",
      "System.Collections.Generic.Dictionary`2"
    ],
    "namespaceMappings": [
      "*, REService"
    ],
    "references": [
      "Microsoft.Bcl.AsyncInterfaces, {Microsoft.Bcl.AsyncInterfaces, 1.1.0}",
      "Microsoft.Bcl.HashCode, {Microsoft.Bcl.HashCode, 1.1.0}",
      "Microsoft.Data.SqlClient, {Microsoft.Data.SqlClient, 1.0.19269.1}",
      "Microsoft.EntityFrameworkCore, {Microsoft.EntityFrameworkCore, 3.1.1}",
      "Microsoft.EntityFrameworkCore.Abstractions, {Microsoft.EntityFrameworkCore.Abstractions, 3.1.1}",
      "Microsoft.EntityFrameworkCore.Relational, {Microsoft.EntityFrameworkCore.Relational, 3.1.1}",
      "Microsoft.EntityFrameworkCore.SqlServer, {Microsoft.EntityFrameworkCore.SqlServer, 3.1.1}",
      "Microsoft.Extensions.Caching.Abstractions, {Microsoft.Extensions.Caching.Abstractions, 3.1.1}",
      "Microsoft.Extensions.Caching.Memory, {Microsoft.Extensions.Caching.Memory, 3.1.1}",
      "Microsoft.Extensions.Configuration, {Microsoft.Extensions.Configuration, 3.1.1}",
      "Microsoft.Extensions.Configuration.Abstractions, {Microsoft.Extensions.Configuration.Abstractions, 3.1.1}",
      "Microsoft.Extensions.Configuration.Binder, {Microsoft.Extensions.Configuration.Binder, 3.1.1}",
      "Microsoft.Extensions.Configuration.FileExtensions, {Microsoft.Extensions.Configuration.FileExtensions, 3.1.1}",
      "Microsoft.Extensions.Configuration.Json, {Microsoft.Extensions.Configuration.Json, 3.1.1}",
      "Microsoft.Extensions.DependencyInjection, {Microsoft.Extensions.DependencyInjection, 3.1.1}",
      "Microsoft.Extensions.DependencyInjection.Abstractions, {Microsoft.Extensions.DependencyInjection.Abstractions, 3.1.1}",
      "Microsoft.Extensions.FileProviders.Abstractions, {Microsoft.Extensions.FileProviders.Abstractions, 3.1.1}",
      "Microsoft.Extensions.FileProviders.Physical, {Microsoft.Extensions.FileProviders.Physical, 3.1.1}",
      "Microsoft.Extensions.FileSystemGlobbing, {Microsoft.Extensions.FileSystemGlobbing, 3.1.1}",
      "Microsoft.Extensions.Logging, {Microsoft.Extensions.Logging, 3.1.1}",
      "Microsoft.Extensions.Logging.Abstractions, {Microsoft.Extensions.Logging.Abstractions, 3.1.1}",
      "Microsoft.Extensions.Options, {Microsoft.Extensions.Options, 3.1.1}",
      "Microsoft.Extensions.Options.ConfigurationExtensions, {Microsoft.Extensions.Options.ConfigurationExtensions, 3.1.1}",
      "Microsoft.Extensions.Primitives, {Microsoft.Extensions.Primitives, 3.1.1}",
      "Newtonsoft.Json, {Newtonsoft.Json, 12.0.3}",
      "System.Collections.Immutable, {System.Collections.Immutable, 1.7.0}",
      "System.ComponentModel.Annotations, {System.ComponentModel.Annotations, 4.7.0}",
      "System.Diagnostics.DiagnosticSource, {System.Diagnostics.DiagnosticSource, 4.7.0}"
    ],
    "targetFramework": "netcoreapp3.1",
    "typeReuseMode": "All"
  }
}

--- a/src/libraries/Microsoft.XmlSerializer.Generator/pkg/build/dotnet-Microsoft.XmlSerializer.Generator.runtimeconfig.json
+++ b/src/libraries/Microsoft.XmlSerializer.Generator/pkg/build/dotnet-Microsoft.XmlSerializer.Generator.runtimeconfig.json
@@ -5,6 +5,6 @@
       "name": "Microsoft.NETCore.App",
       "version": "2.0.0"
     },
-    "rollForward": "Major"
+    "rollForward": "LatestMajor"
   }
 }

